fe_utils/solvers/helmholtz.py

```diff
@@ -28,6 +32,27 @@ def assemble(fs, f):
     l = np.zeros(fs.node_count)
 
     # Now loop over all the cells and assemble A and l
+    for c in range(mesh.entity_counts[-1]):
+        # Find the appropriate global node numbers for this cell.
+        nodes = fs.cell_nodes[c, :]
+
+        # Compute the change of coordinates.
+        J = mesh.jacobian(c)
+        detJ = np.abs(np.linalg.det(J))
+
+        # Compute the actual cell quadrature for rignt-hand side
+        l[nodes] += np.dot( phi.T * np.dot(f.values[nodes], phi.T), Q.weights) * detJ
+
+        '''
+        for i in range(nodes.shape[0]):
+                  lm[nodes[i]] += np.dot( phi[:,i].T * np.dot(f.values[nodes], phi.T), Q.weights) * detJ
+        '''
+        
+        # Compute the actual cell quadrature for left-hand side
+
+        for q in range(Q.weights.shape[0]):
+            A[np.ix_(nodes, nodes)] += np.dot( np.dot(np.dot(np.linalg.inv(J.T), psi[0].T).T, np.dot(np.linalg.inv(J.T), psi[0].T)) +  np.dot(phi[q,:].T, phi[q,:]), Q.weights[q]) * detJ 
```

Comment:
As we discussed, I think this would be a lot easier to understand if it used `np.einsum`